### PR TITLE
`constrained-generators`: clean up interface

### PR DESCRIPTION
**Pull request by MaximilianAlgehed**
# Description

Removes some TODOs in the interface and cleans things up a bit to reduce the scope for introducing breaking changes later on.

# Checklist

- [x] Commit sequence broadly makes sense and commits have useful messages
- [x] New tests are added if needed and existing tests are updated
- [x] When applicable, versions are updated in `.cabal` and `CHANGELOG.md` files according to the
      [versioning process](https://github.com/intersectmbo/cardano-ledger/blob/master/RELEASING.md#versioning-process).
- [x] The version bounds in `.cabal` files for all affected packages are updated. **If you change the bounds in a cabal file, that package itself must have a version increase.** (See [RELEASING.md](https://github.com/intersectmbo/cardano-ledger/blob/master/RELEASING.md#versioning-process))
- [x] All visible changes are prepended to the latest section of a `CHANGELOG.md` for the affected packages. **New section is never added with the code changes.** (See [RELEASING.md](https://github.com/intersectmbo/cardano-ledger/blob/master/RELEASING.md#changelogmd))
- [x] Code is formatted with [`fourmolu`](https://github.com/fourmolu/fourmolu) (use `scripts/fourmolize.sh`)
- [x] Cabal files are formatted (use `scripts/cabal-format.sh`)
- [x] [`hie.yaml`](https://github.com/intersectmbo/cardano-ledger/blob/master/hie.yaml) has been updated (use `scripts/gen-hie.sh`)
- [x] Self-reviewed the diff

### CODE REVIEW

File: libs/constrained-generators/src/Constrained.hs [hunk 1]
```diff
@@ -112,13 +111,8 @@ module Constrained (
   shrinkWithNumSpec,
   conformsToNumSpec,
   toPredsNumSpec,
-  -- TODO: this is super yucky, it would be good to implement
-  -- the linear lambda thing to get rid of this.
```
Comment (teodanciu) [inline libs/constrained-generators/src/Constrained.hs:116]: I don't know what this means but I support the principle: the fewer super yucky things the better!

**Review by teodanciu — APPROVED** [review]
Looks good to me

**Review by teodanciu — COMMENTED** [review]
I guess we're not yet releasing this on Chap, so no need to increment the version (as it would the case if we releasing)?

**Comment by MaximilianAlgehed** [comment]
> I guess we're not yet releasing this on Chap, so no need to increment the version (as it would the case if we releasing)?

Indeed. I'm hoping to clean as much of this stuff up as possible before we release on chap.

**Pull request merged**